File: JavaScript/DailyCodeChallenge/260301_isFlat.js
function isFlat(arr) {

  const answer =  !arr.some(item => Array.isArray(item))
  console.log(answer)
  return answer

}


isFlat([1, 2, 3, 4])
isFlat([1, [2, [3, [4, [5]]]], 6])

/*
Flattened
Given an array, determine if it is flat.

An array is flat if none of its elements are arrays.

Tests
Passed:1. isFlat([1, 2, 3, 4]) should return true.
Passed:2. isFlat([1, [2, 3], 4]) should return false.
Passed:3. isFlat([1, 0, false, true, "a", "b"]) should return true.
Passed:4. isFlat(["a", [0], "b", true]) should return false.
Passed:5. isFlat([1, [2, [3, [4, [5]]]], 6]) should return false.
*/

File: Python/Daily_Code_Challenge/260301-is-flat.py
def is_flat(arr):

    answer = [isinstance(item,list) for item in arr]
    print(answer)

    print(not True in answer)
    return not True in answer



is_flat(["a", [0], "b", True])

'''
Flattened
Given an array, determine if it is flat.

An array is flat if none of its elements are arrays.

Tests
Passed:1. is_flat([1, 2, 3, 4]) should return True.
Passed:2. is_flat([1, [2, 3], 4]) should return False.
Passed:3. is_flat([1, 0, False, True, "a", "b"]) should return True.
Passed:4. is_flat(["a", [0], "b", True]) should return False.
Passed:5. is_flat([1, [2, [3, [4, [5]]]], 6]) should return False.

'''